src/mmrelay/k8s_utils.py

```
@@ -0,0 +1,512 @@
+        variables (dict): Dictionary of variable names to values. Placeholders
+            on their own line are treated as block substitutions and will
+            inherit the line indentation.
+
+    Returns:
+        str: Rendered template
+    """
+    placeholders = set(_PLACEHOLDER_RE.findall(template))
+    missing = sorted(placeholders - set(variables.keys()))
+    if missing:
+        missing_vars = ", ".join(missing)
+        raise ValueError(f"Missing template variables: {missing_vars}")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)